### PR TITLE
#4649 

### LINKED ISSUE #4649: Broken Search
### Problem
- Broken Search

build docs with: $make html works as a charm. Pointing docs via webserver is working fine, but search doesn't works:
- cannot find any numbers: index.rst contain "1234567", which is displayed on browser,but if I search it no results displayed. This happen also with IP Address, and others numbers;
- Lots of words with dots doesn't appear: if I search www.example.com, search doesn't find it.

#### Error logs / results
No errors displayed

#### Expected results
Search results.

### Reproducible project / your project
Private project.

### Environment info
- OS: Linux Ubuntu 14.04.5
- Python version: 2.7.6
- Sphinx version: 1.7.0
- same behaviour in multiple browsers.

### PR DESCRIPTION
**Pull request by TimKam**
Subject: Fix: search always returns nothing when one term is shorter than 3

### Feature or Bugfix
- Bugfix

### Purpose
The search function considers every page a "miss"/no match if the search contains a term with length <= 2; 
This is problem when users search for filenames, identifiers, et cetera.
This PR changes the behavior to be more lenient in such cases: if an additional search term is of length <= 2, it is ignored (no longer a necessary condition for a match).

### Relates
- Closes #4649

### CODE REVIEW

File: sphinx/themes/basic/static/searchtools.js [hunk 1]
```diff
@@ -428,7 +428,11 @@ var Search = {
       var valid = true;
 
       // check if all requirements are matched
-      if (fileMap[file].length != searchterms.length)
+      var filteredTerms = // as search terms with length < 3 are discarded: ignore
```
Comment (tk0miya) [inline sphinx/themes/basic/static/searchtools.js]: I think this value is a count of terms. How about `filteredTermCount`?

File: sphinx/themes/basic/static/searchtools.js [hunk 2]
```diff
@@ -428,7 +428,11 @@ var Search = {
...
+        searchterms.filter(function(term){return term.length > 2}).length
+      var allRequirementsMatched = fileMap[file].length != searchterms.length &&
```
Comment (tk0miya) [inline sphinx/themes/basic/static/searchtools.js]: IMO, this variable is not needed. I prefer to implement this as simple if-condition. What do you think?

**Review by tk0miya — COMMENTED** [review]
LGTM with nits

File: sphinx/themes/basic/static/searchtools.js (hunk 2 above)
Comment (TimKam) [reply sphinx/themes/basic/static/searchtools.js]: I agree. It's more concise without this assignment. I will work in the feedback and merge when the CI is green.

**Pull request merged**